autogpt_platform/backend/backend/blocks/mem0.py

```diff
@@ -0,0 +1,310 @@
+        memory: str = SchemaField(description="Memory created")
+        error: str = SchemaField(description="Error message if operation fails")
+
+    def __init__(self):
+        super().__init__(
+            id="dce97578-86be-45a4-ae50-f6de33fc935a",
+            description="Add new memories to Mem0 with user segmentation",
+            input_schema=AddMemoryBlock.Input,
+            output_schema=AddMemoryBlock.Output,
+            test_input={
+                "content": [{"role": "user", "content": "I'm a vegetarian"}],
+                "metadata": {"food": "vegetarian"},
+                "credentials": TEST_CREDENTIALS_INPUT,
+            },
+            test_output=[("action", "NO_CHANGE")],
+            test_credentials=TEST_CREDENTIALS,
+            test_mock={"_get_client": lambda credentials: MockMemoryClient()},
+        )
+
+    def run(
+        self,
+        input_data: Input,
+        *,
+        credentials: APIKeyCredentials,
+        user_id: str,
+        graph_id: str,
+        run_id: str,
+        **kwargs
+    ) -> BlockOutput:
+        try:
+            client = self._get_client(credentials)
+
+            # Convert input to messages format if needed
+            messages = (
+                input_data.content
+                if isinstance(input_data.content, list)
+                else [{"role": "user", "content": input_data.content}]
+            )
+
+            params = {
+                "user_id": user_id,
+                "output_format": "v1.1",
+                "metadata": input_data.metadata,
+            }
+
+            if input_data.limit_memory_to_run:
+                params["run_id"] = run_id
+            if input_data.limit_memory_to_agent:
+                params["agent_id"] = graph_id
+
+            # Use the client to add memory
+            result = client.add(
+                messages,
+                **params,
+            )
+
+            if len(result.get("results", [])) > 0:
+                for result in result.get("results", []):
+                    yield "action", result["event"]
+                    yield "memory", result["memory"]
+            else:
+                yield "action", "NO_CHANGE"
+
+        except Exception as e:
+            yield "error", str(object=e)
+
+
+class SearchMemoryBlock(Block, Mem0Base):
+    """Block for searching memories in Mem0"""
+
+    class Input(BlockSchema):
+        credentials: CredentialsMetaInput[
+            Literal[ProviderName.MEM0], Literal["api_key"]
+        ] = CredentialsField(description="Mem0 API key credentials")
+        query: str = SchemaField(
+            description="Search query",
+            advanced=False,
+        )
+        trigger: bool = SchemaField(
+            description="An unused field that is used to (re-)trigger the block when you have no other inputs",
+            default=False,
+            advanced=False,
+        )
+        categories_filter: list[str] | None = SchemaField(
```

Comment:
Does it need to be none or we can just set [] as default.
If this is only for the same of making it non advance, we can set advanced=False

Reply:
fixed by making advanced

Reply:
fixed